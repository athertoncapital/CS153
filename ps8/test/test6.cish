main() {
	a = 1;
	b = 2;
	c = 3;
	d = 4;
	out = 0;
	if (a > 0) {
		out = out;
		if (b > 0) {
			c = 3;
			out = out + b;
		}
		else {
			if (c > 0) {
				c = a + c;
			}
		}
	}
	else {
		if (a > d) {
			d = a + c;
		}
	}
	out = d;
	return out;
}

	
		
				
