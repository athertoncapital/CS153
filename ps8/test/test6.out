main:
x0 := $30
x1 := $31
x2 := $16
x3 := $17
x4 := $18
x5 := $19
x6 := $20
x7 := $21
x8 := $22
x9 := $23
a := 1
b := 2
c := 3
d := 4
out := 0
if a>0 goto .L3 else goto .L2
.L3:
out := out
if b>0 goto .L6 else goto .L5
.L6:
c := 3
x10 := out+b
out := x10
jump .L4
.L5:
if c>0 goto .L9 else goto .L8
.L9:
x11 := a+c
c := x11
jump .L7
.L8:
jump .L7
.L7:
jump .L4
.L4:
jump .L1
.L2:
if a>d goto .L12 else goto .L11
.L12:
x12 := a+c
d := x12
jump .L10
.L11:
jump .L10
.L10:
jump .L1
.L1:
out := d
$2 := out
jump .L0
.L13:
jump .L0
.L0:
$30 := x0
$31 := x1
$16 := x2
$17 := x3
$18 := x4
$19 := x5
$20 := x6
$21 := x7
$22 := x8
$23 := x9
return

main() {
   a = 1;
   b = 2;
   c = 3;
   d = 4;
   out = 0;
   if (a>0) {
     out = out;
     if (b>0) {
       c = 3;
       out = out+b;
     } else {
       if (c>0) {
         c = a+c;
       } else {
         0;
       }
     }
   } else {
     if (a>d) {
       d = a+c;
     } else {
       0;
     }
   }
   out = d;
   return out;
}

graph g {
  a -- b;
  a -- c;
  a -- d;
  a -- out;
  b -- c;
  b -- d;
  b -- out;
  c -- d;
  c -- out;
  d -- out;
  d -- x0;
  d -- x1;
  d -- x2;
  d -- x3;
  d -- x4;
  d -- x5;
  d -- x6;
  d -- x7;
  d -- x8;
  d -- x9;
  out -- x0;
  out -- x1;
  out -- x2;
  out -- x3;
  out -- x4;
  out -- x5;
  out -- x6;
  out -- x7;
  out -- x8;
  out -- x9;
  x0 -- x1;
  x0 -- x2;
  x0 -- x3;
  x0 -- x4;
  x0 -- x5;
  x0 -- x6;
  x0 -- x7;
  x0 -- x8;
  x0 -- x9;
  x1 -- x2;
  x1 -- x3;
  x1 -- x4;
  x1 -- x5;
  x1 -- x6;
  x1 -- x7;
  x1 -- x8;
  x1 -- x9;
  x2 -- x3;
  x2 -- x4;
  x2 -- x5;
  x2 -- x6;
  x2 -- x7;
  x2 -- x8;
  x2 -- x9;
  x3 -- x4;
  x3 -- x5;
  x3 -- x6;
  x3 -- x7;
  x3 -- x8;
  x3 -- x9;
  x4 -- x5;
  x4 -- x6;
  x4 -- x7;
  x4 -- x8;
  x4 -- x9;
  x5 -- x6;
  x5 -- x7;
  x5 -- x8;
  x5 -- x9;
  x6 -- x7;
  x6 -- x8;
  x6 -- x9;
  x7 -- x8;
  x7 -- x9;
  x8 -- x9;
}

